scripts/superdesk/directives/sdTypeahead.js

```diff
@@ -67,6 +67,11 @@
                             $scope.hide = true;
                             $scope.focused = false;
                             $scope.select({item: item});
+                            $scope.active = null;
+                            $scope.term = null;
+                            if (item) {
+                                $document.triggerHandler('click');
```

Comment:
Previous when you click on some item, or press enter, active element remained there (wasn't reset) so if we remove that item and then press enter to open dropdown it will add previously selected item although we just pressed enter to open dropdown.

`$scope.term=null` resets search, so on next open of dropdown it will be emtpy.

And third part when you open dropdown, type something  and press enter, it wasn't closing dropdown. I tried, but couldn't resolve it different, it's a bit tricky..